### DISCUSSION TITLE
How to add SPDM extension field to Caliptra's IDevID cert?

### DISCUSSION THREAD

**Discussion by yh36** [chipsalliance/caliptra-sw#1690] [Q&A]
In SPDM DSP0274, the [Hardware identity OID](https://github.com/DMTF/SPDM-WG/blob/master/docs/DSP0274/SPDMSpecification.md#hardware-identity-oid) is used to identify the hardware identity certificate in a cert chain. Per the [IDevID certificate format](https://github.com/chipsalliance/Caliptra/blob/main/doc/Caliptra.md#idevid-certificate), Caliptra doesn't support this OID. How does Caliptra satisfy the SPDM requirement if customers want this DMTF SPDM extension field?

![image](https://github.com/user-attachments/assets/f763dd23-91f9-4dc1-9a28-2841cc293c4d)

**Comment by ballle98** [comment]
see #1885

**Comment by jhand2** [comment]
For IDevID specifically, Caliptra only creates a CSR. The vendor CA is free to add this extension to the IDevID certificate.
